infrastructure/csv/client.go

```diff
@@ -0,0 +1,78 @@
+package csv
+
+import (
+	"encoding/csv"
+	"fmt"
+	"github.com/esvarez/go-api/internal/boardgame"
+	errs "github.com/esvarez/go-api/pkg/error"
+	"log"
+	"os"
+	"strconv"
+)
+
+const (
+	idI = iota
+	nameI
+	descriptionI
+	minPlayersI
+	maxPlayersI
+	DurationI
+)
+
+type Client struct {
+	path string
+}
+
+func NewCSVClient(filePath string) *Client {
+	getCSVReader(filePath)
+	return &Client{
+		path: filePath,
+	}
+}
+
+func getCSVReader(filePath string) *csv.Reader {
+	f, err := os.Open(filePath)
+	if err != nil {
+		log.Fatalf("failed opening file: %s", err)
+	}
+
+	return csv.NewReader(f)
+}
+
+func (c *Client) FindBoardGame(id int) (*boardgame.BoardGame, error) {
```

Comment:
No es necesario usar punteros cuando se manejan estructuras salvo algunos casos.

https://medium.com/@meeusdylan/when-to-use-pointers-in-go-44c15fe04eac